ledger/onlineaccountscache.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+// Worst case memory usage = 2500 * 320 * 150B = 120MB
+const onlineAccountsCacheMaxSize = 2500
+
+type onlineAccountsCache struct {
+	// each persistedOnlineAccountDataList stores online account data with newest
+	// at the front, and oldest at the back.
+	accounts map[basics.Address]*persistedOnlineAccountDataList
```

Comment:
I think we should only store round and account data instead of the full `persistedOnlineAccountDataList`.

Reply:
done